### PR TITLE
[RFC][NI] refactored alert, added fixed and close behaviour

### PR DESCRIPTION
**Pull request by telmaantunes**
screen shot 2017-03-15 at 17 58 42

- on scroll:
screen shot 2017-03-15 at 17 58 54

### CODE REVIEW

File: addon/components/uni-alert.js [hunk 1]
```diff
@@ -23,5 +28,58 @@ export default Component.extend({
...
+    }
+
+    return 'alert';
+  }),
+
+  init() {
+    this._super();
+
+    if (this.get('isFixed')) {
+      $(window).bind('scroll', () => {
```
Comment (montoias) [inline addon/components/uni-alert.js]: make sure to unbind also after the component is destroyed ;)

File: addon/templates/components/uni-alert.hbs [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+		<div class="uni-alert__content">{{subtitle}}</div>
   {{/if}}
 
-  <div class="alert__content">
+  <div class="uni-alert__content">
     {{yield}}
   </div>
+
+  {{#if hasClose}}
+		<div class="uni-alert__close" {{action 'close'}}>{{inline-svg "icons/cross.svg"}}</div>
```
Comment (montoias) [inline addon/templates/components/uni-alert.hbs]: tab

File: addon/templates/components/uni-alert.hbs [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
-  {{else}}
-    {{inline-svg "icons/icon-warning.svg"}}
+{{inline-svg (i "icons/${iconName}.svg" iconName=iconName)}}
+
+<div class="uni-alert__wrapper">
+	{{#if title}}
+		<div class="uni-alert__title">{{title}}</div>
+  {{/if}}
+  {{#if subtitle}}
+		<div class="uni-alert__content">{{subtitle}}</div>
```
Comment (montoias) [inline addon/templates/components/uni-alert.hbs]: tab

File: addon/templates/components/uni-alert.hbs [hunk 4]
```diff
@@ -1,11 +1,18 @@
-<div class="alert__wrapper">
-  {{#if isSuccess}}
-    {{inline-svg "icons/success-icon.svg"}}
-  {{else}}
-    {{inline-svg "icons/icon-warning.svg"}}
+{{inline-svg (i "icons/${iconName}.svg" iconName=iconName)}}
+
+<div class="uni-alert__wrapper">
+	{{#if title}}
+		<div class="uni-alert__title">{{title}}</div>
```
Comment (montoias) [inline addon/templates/components/uni-alert.hbs]: tab

File: addon/components/uni-alert.js [hunk 5]
```diff
@@ -2,19 +2,24 @@ import Ember from 'ember';
...
 import UniAlertTypes from 'ember-cli-uniq/enums/uni-alert-type';
 
-const { Component, computed  } = Ember;
+const { Component, computed, $ } = Ember;
 
 export default Component.extend({
   classNames: ['uni-alert'],
   classNameBindings: [
-    'isFixed:uni-alert--fixed',
+    'isSticky:uni-alert--fixed',
```
Comment (montoias) [inline addon/components/uni-alert.js]: why change the name?

File: addon/components/uni-alert.js [hunk 6]
```diff
@@ -2,19 +2,24 @@ import Ember from 'ember';
...
     'isSuccess:uni-alert--success',
     'isError:uni-alert--error',
-    'isWarning:uni-alert--warning'
+    'isWarning:uni-alert--warning',
+    'isClosed:uni-alert--closed'
   ],
   layout,
 
   isFixed: false,
```
Comment (montoias) [inline addon/components/uni-alert.js]: not sure why keeping the isFixed

**Review by montoias — CHANGES_REQUESTED** [review]

File: addon/components/uni-alert.js (hunk 1 above)
Comment (telmaantunes) [reply addon/components/uni-alert.js]: done

**Pull request merged**